File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        numset = set(nums)
        longest = 0

        for num in numset:
            if num - 1 not in numset:
                length = 1
                current = num + 1

                while current in numset:
                    length += 1
                    current += 1

                longest = max(longest, length)

        return longest


        

File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> curr;
        curr.push_back({});
        for (int i = 0; i < nums.size(); i++) {
            int count = 0;
            int currSize = curr.size();
            while (count < currSize) {
                vector<int> currVec = curr[count]; // access current vector
                currVec.push_back(nums[i]); //append current num to current vector
                curr.push_back(currVec); //push back new vector onto result curr list
                count++;
            }    
        }
        return curr;
    }
};
